apps/admin_app/lib/admin_app_web/controllers/dashboard_controller.ex

```diff
@@ -0,0 +1,96 @@
+defmodule AdminAppWeb.DashboardController do
+  use AdminAppWeb, :controller
+
+  alias Snitch.Data.Model
+
+  def index(conn, params) do
+    {start_date, end_date} = get_date_from_params(params)
+
+    order_state_counts = get_order_state_count(start_date, end_date)
+    product_state_counts = get_product_state_count(start_date, end_date)
+    order_datepoints = get_order_datapoints(start_date, end_date)
+    payment_datapoints = get_payment_datapoints(start_date, end_date)
+
+    render(conn, "index.html",
+      order_state_counts: order_state_counts,
+      product_state_counts: product_state_counts,
+      order_datepoints: order_datepoints,
+      payment_datapoints: payment_datapoints
+    )
+  end
+
+  defp get_naive_date_time(date) do
+    Date.from_iso8601(date)
+    |> elem(1)
+    |> NaiveDateTime.new(~T[00:00:00])
+    |> elem(1)
+  end
+
+  defp get_date_from_params(params) do
+    today = Date.utc_today() |> Date.to_string() |> get_naive_date_time()
+
+    start_date =
+      case Map.get(params, "from") do
+        nil -> today
+        "" -> today
+        date -> date |> get_naive_date_time()
+      end
+
+    end_date =
+      case Map.get(params, "to") do
+        nil -> today
+        "" -> today
+        date -> date |> get_naive_date_time()
+      end
+
+    {start_date, end_date}
+  end
+
+  defp get_order_state_count(start_date, end_date) do
+    Model.Order.get_order_count_by_state(start_date, end_date)
+  end
+
+  defp get_product_state_count(start_date, end_date) do
+    Model.Product.get_product_count_by_state(start_date, end_date)
+  end
+
+  @spec get_order_datapoints(any(), any()) :: %{data: any(), labels: any()}
+  defp get_order_datapoints(start_date, end_date) do
+    Model.Order.get_order_count_by_date(start_date, end_date)
+    |> format_response()
+  end
+
+  defp get_payment_datapoints(start_date, end_date) do
+    Model.Payment.get_payment_count_by_date(start_date, end_date)
```

Comment:
the pipe should start with a raw value, I think credo would have given some warning here, let's not add more to already existing many credo issues.

Reply:
Will this work, if I make this without pipe syntax, or there is something else for it that I am missing.